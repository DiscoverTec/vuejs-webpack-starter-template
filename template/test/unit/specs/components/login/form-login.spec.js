import Vuex from 'vuex'
import { createRenderer } from 'vue-server-renderer'
import { shallow, createLocalVue } from '@vue/test-utils'

import Component from '@/components/login/form-login'

const localVue = createLocalVue()
localVue.use(Vuex)

describe(`Component ${Component.name}`, () => {

  it('should display loading and messages', () => {
    const renderer = createRenderer()
    const wrapper = shallow(Component, { localVue })
    wrapper.vm.loading = true
    wrapper.vm.message = 'Sample Message'
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })

  it('should getToken for the user', async () => {
    const renderer = createRenderer()
    let $router = []
    let actions = {
      getToken: jest.fn(),
      veeValidateAll: () => true
    }
    let store = new Vuex.Store({
      modules: {
        authorization: {
          namespaced: true,
          actions
        }
      }
    })
    const wrapper = shallow(Component, { localVue, store, mocks: { $router } })
    wrapper.vm.message = 'this message should clear'
    await wrapper.find('form').trigger('submit')
    expect(wrapper.vm.message).toEqual('')
    expect(actions.getToken.mock.calls).toHaveLength(1)
  })

  it('should display error when 500', async () => {
    const renderer = createRenderer()
    let actions = {
      veeValidateAll: () => true,
      getToken: () => { throw { response: { status: 500 } } }
    }
    let store = new Vuex.Store({
      modules: {
        authorization: {
          namespaced: true,
          actions
        }
      }
    })
    const wrapper = shallow(Component, { localVue, store })
    await wrapper.find('form').trigger('submit')
    expect(wrapper.vm.message).toMatchSnapshot()
  })

  it('should display error when 403', async () => {
    const renderer = createRenderer()
    let actions = {
      veeValidateAll: () => true,
      getToken: () => { throw { response: { status: 403 } } }
    }
    let store = new Vuex.Store({
      modules: {
        authorization: {
          namespaced: true,
          actions
        }
      }
    })
    const wrapper = shallow(Component, { localVue, store })
    await wrapper.find('form').trigger('submit')
    expect(wrapper.vm.message).toMatchSnapshot()
  })
})