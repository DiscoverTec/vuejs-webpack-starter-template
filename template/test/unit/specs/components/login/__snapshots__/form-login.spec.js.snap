// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component form-login should display error when 403 1`] = `"Incorrect username or password"`;

exports[`Component form-login should display error when 500 1`] = `"Error: Unable to login"`;

exports[`Component form-login should display loading and messages 1`] = `
<form>
    <!---->
    <!---->
    <button type="submit" disabled="disabled" class="btn btn-success btn-block">
        Log In
        <i aria-hidden="true" class="fa fa-spinner fa-spin"></i>
    </button>
    <div class="alert alert-warning mt-2">
        Sample Message
    </div>
</form>
`;
